src/NUnitFramework/framework/Attributes/TestFixtureSourceAttribute.cs

```diff
@@ -123,8 +124,16 @@ public IEnumerable<TestSuite> BuildFrom(ITypeInfo typeInfo, IPreFilter filter)
         {
             Type sourceType = SourceType ?? typeInfo.Type;
 
+            var fixtureSuite = new ParameterizedFixtureSuite(typeInfo);
+            fixtureSuite.ApplyAttributesToTest(typeInfo.Type.GetTypeInfo());
+
             foreach (ITestFixtureData parms in GetParametersFor(sourceType))
-                yield return _builder.BuildFrom(typeInfo, filter, parms);
+            {
+                TestSuite fixture = _builder.BuildFrom(typeInfo, filter, parms);
+                fixtureSuite.Add(fixture);
+            }
+
+            yield return fixtureSuite;
```

Comment:
Instead of doing this inside TestFixtureSourceAttribute, could you implement it in the caller of this method, `DefaultSuiteBuilder.BuildFrom`? I'm thinking that way the fix applies to all IFixtureBuilder attributes and not just TestFixtureSourceAttribute. I think this might automatically fix it for TestFixtureAttribute without changes too.